_posts/2018-02-16-Java-Time2.md

```diff
@@ -0,0 +1,1564 @@
+            zone.createChild(subnet); // 11
+        } catch (RestException e) {
+            e.printStackTrace();
+        }
+        
+        return subnet;
+    }
+    
+    public List<Subnet> fetchAllSubnetsForZone(Zone zone) {
+        SubnetsFetcher fetcher = zone.getSubnets(); // 12
+        List<Subnet> list = new ArrayList<>();
+
+        try {
+            if (fetcher.count() > 0) { // 13
+                list = fetcher.get(); // 14
+            }
+        } catch (RestException e) {
+            e.printStackTrace();
+        }
+
+        return list;
+    }
+    
+    public void deleteSubnet(Subnet subnet) { // 15
+        if (subnet == null) {
+            System.out.println("Subnet does not exist");
+        } else {
+            System.out.println("Subnet " + subnet.getName() + " will be deleted");
+            try {
+                subnet.delete(); // 16
+            } catch (RestException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+}
+{% endhighlight %}
+
+In Line 1 of the `verifySubnet` method, we supply arguments for the parent Zone object and the Subnet attributes.
+
+Line 2 checks whether an existing Subnet object matches the name that is passed in via `NetworkDescriptor`. If the Subnet already exists, then we return that Subnet object.  If the Subnet does not exist, then in Line 3 we call the `createSubnet` method to create and return a new SUbnet object.
+
+Line 4 of the `fetchSubnet` method sets up a search filter based on the name of our subnet.  In Line 5, we create an instance of `SubnetsFetcher`.  Then Line 6 retrieves the Subnet object matching the name in the search filter. 
+
+In Line 7 of the `createSubnet` method, we call the constructor of the VSPK Subnet class to create a brand new Zone object and then in Lines 8 through 10 we assign a few attributes for our new subnet.  Finally, in Line 11, we attach our new Subnet object to the already existing Zone object.
+
+Lines 12 to 16 of methods `fetchAllSubnetsForZone` and `deleteSubnet` will be covered later when we discuss how to remove our network layout from the VSD.
+
+# Adding the Subnet to our Network
+
+Here is the updated version of the `NetworkBuilder` class that integrates with `NetworkSubnetHandler`.  There are 2 minor changes as shown below.  At Line 1, we instantiate the new handler and at Line 2, we call its `verifySubnet` method and pass it our existing Zone object and the `NetworkDescriptor` object which contains the name and IP attributes of our subnet.
+
+{% highlight java %}
+import java.util.List;
+import net.nuagenetworks.bambou.RestException;
+import net.nuagenetworks.vspk.v5_0.Subnet;
+import net.nuagenetworks.vspk.v5_0.Zone;
+import net.nuagenetworks.vspk.v5_0.Domain;
+import net.nuagenetworks.vspk.v5_0.DomainTemplate;
+import net.nuagenetworks.vspk.v5_0.Enterprise;
+import net.nuagenetworks.vspk.v5_0.VSDSession;
+
+public class NetworkBuilder {
+    private String url = "https://135.228.4.108:8443";

```

Comment:
Don't use a 135.... IP in documentation, use one from https://tools.ietf.org/html/rfc5737

Suggestion: `198.51.100.108`